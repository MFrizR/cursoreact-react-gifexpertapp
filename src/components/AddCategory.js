import React, { useState } from 'react';
import PropsTypes from 'prop-types';

const AddCategory = ({ setCategories } ) => {

    const [inputValue, setInputValue] = useState(' ');

    const handleInputChange = (e) => {
        console.log(e.target.value);
        setInputValue (e.target.value)
    }

    const handleSubmit = (e) => {
        // Prevenir comportamiento por defecto del formulario
        e.preventDefault();
        

        if(inputValue.trim().length > 2){
            setCategories( cats => [  inputValue,...cats]);
            setInputValue('');
        }
        console.log('Submit Hecho');
    }

    return (
        <>
        <form onSubmit = { handleSubmit }>
            <input
                type= "text"
                value= { inputValue }
                onChange = { handleInputChange }
            />
        </form>
          

          
        </>
    )
}

AddCategory.propsTypes = {
    setCategories: PropsTypes.func.isRequired
};

 AddCategory.defaultProps = {

    //setCategories: 
};  

export default AddCategory
