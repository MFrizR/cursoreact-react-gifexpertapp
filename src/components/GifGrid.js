//import React, { useState, useEffect } from 'react'
import React from 'react';
import PropsTypes from 'prop-types'
import { GifGridItem } from './GifGridItem';
//import { getGifs } from './helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ( {category} ) => {


    const {data, loading} = useFetchGifs( category );  

//console.log(data)
//const [count, setCount] = useState(0);

/* const [images, setImages] = useState([]);


 */


//getGifs();


    return (
        <>
        <h3>{ category }</h3>
         { loading && <p>Loading</p> } 
        <div className="card-grid">
           

            {
                data.map (( img ) => (

                    <GifGridItem 
                        { ...img }
                        key = { img.id }
                    />
                ))
            }

          {/*   <ol>
                {
                    images.map( img => (
                        <li key={ img.id }>{ img.title }</li>
                    ))
                }
            </ol> */}
            {/* <h3>{ count }</h3> */}
            {/* <button onClick={ () => setCount( count +1 )}></button> */}
         </div>
        </>
    )
}

GifGrid.PropsTypes = {

    category: PropsTypes.string.isRequired
}
