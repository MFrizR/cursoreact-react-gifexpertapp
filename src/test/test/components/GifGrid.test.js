import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom'
import { GifGrid } from '../../../components/GifGrid';
import { useFetchGifs } from '../../../hooks/useFetchGifs';
jest.mock('../../../hooks/useFetchGifs');

describe('Pruebas en el <GifGrid/>', () => {
    
    const category = 'One Punch';

    test('1.- Debe de mostrarse correctamente ', () => {
    
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true

        });

        const wrapper = shallow(<GifGrid category= { category }/>);
        expect( wrapper ).toMatchSnapshot();
    })

    test('2.- Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
       
        const gifs = [{
            id: 'ABC',
            url: 'https://local',
            title: 'cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false

        });

        const wrapper = shallow(<GifGrid category= { category }/>)
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists()).toBe(false);
        expect( wrapper.find('GifGridItem').length).toBe( gifs.length );
    })
    
    
})
