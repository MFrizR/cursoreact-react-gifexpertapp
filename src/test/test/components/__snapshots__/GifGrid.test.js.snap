// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid/> 1.- Debe de mostrarse correctamente  1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el <GifGrid/> 2.- Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://local"
    />
  </div>
</Fragment>
`;
